src/build.cc

```diff
@@ -252,28 +239,27 @@ string BuildStatus::FormatProgressStatus(
 
         // Overall finished edges per second.
       case 'o':
-        overall_rate_.UpdateRate(finished_edges_);
-        SnprintfRate(overall_rate_.rate(), buf, "%.1f");
+        SnprintfRate(finished_edges_ / (time_millis_ / 1e3), buf, "%.1f");
         out += buf;
         break;
 
         // Current rate, average over the last '-j' jobs.
       case 'c':
-        current_rate_.UpdateRate(finished_edges_);
+        current_rate_.UpdateRate(finished_edges_, time_millis_);
         SnprintfRate(current_rate_.rate(), buf, "%.1f");
         out += buf;
         break;
 
         // Percentage
-      case 'p':
-        percent = (100 * finished_edges_) / total_edges_;
+      case 'p': {
+        int percent = (100 * finished_edges_) / total_edges_;
```

Comment:
Not mission critical, but personally I'd drop a comment near the top of this function that indicates that integer division is, in fact, the intended type of division.